internal/exec/aws_auth.go

```diff
@@ -0,0 +1,209 @@
+	if AwsAuth == nil {
+		return errors.New("no auth config found")
+	}
+	log.Debug("AwsAuth Config", "auth", AwsAuth)
+
+	profile, err := flags.GetString("profile")
+	if err != nil {
+		return err
+	}
+	if profile == "" {
+
+		// Simple Picker
+		items := []string{}
+		for k := range AwsAuth {
+			items = append(items, k)
+		}
+		choose, err := picker.NewSimplePicker("Choose an AwsAuth Config", items).Choose()
+		if err != nil {
+			return err
+		}
+		log.Info("Selected profile", "profile", choose)
+		profile = choose
+	}
+
+	config := AwsAuth[profile]
+	// TODO validate config
+	return ExecuteAwsAuth(profile, config)
+}
+
+func ExecuteAwsAuth(alias string, config schema.AwsAuthConfig) error {
+	ctx := context.Background()
+	store := authstore.NewKeyringAuthStore()
+	keyringKey := fmt.Sprintf("%s-%s", alias, config.Profile)
+	cfg, err := awsconfig.LoadDefaultConfig(ctx, awsconfig.WithRegion(config.Region))
+	if err != nil {
+		panic(fmt.Errorf("failed to load config: %w", err))
+	}
+	ssoClient := sso.NewFromConfig(cfg)
+
+	var credentials *authstore.AuthCredential
+	// 1. Log into Method - Perhaps we already have a valid token
+	cred, err := store.Get(keyringKey)
+	if err == nil && store.IsValid(cred) {
+		// 1.A Valid Token, proceed.
+		log.Info("Valid token found", "alias", alias)
+		credentials = cred
+	} else {
+		// 1.B No valid token, log in
+		// 1.B.1. Start SSO flow
+		tokenOut := auth.SsoSync(config.StartUrl, config.Region)
+		accessToken := *tokenOut.AccessToken
+		log.Info("✅ Logged in! Token acquired.")
+
+		newCred := &authstore.AuthCredential{
+			Method:    authstore.MethodSSO,
+			Token:     accessToken,
+			ExpiresAt: time.Now().Add(time.Duration(tokenOut.ExpiresIn) * time.Second),
+		}
+
+		credentials = newCred
+		err = store.Set(keyringKey, newCred)
+		if err != nil {
+			return fmt.Errorf("failed to save token: %w", err)
+		}
+	}
+
+	// 2. Authenticate and store to ~/.aws/credentials
+	accounts, err := getAccountsInfo(ctx, ssoClient, credentials.Token)
+	if err != nil {
+		return fmt.Errorf("failed to get accounts info: %w", err)
+	}
+
+	var accountID string
+	// TODO if no auto login account is specified, prompt the user to select one
+	for i := 0; i < len(accounts); i++ {
+		if *accounts[i].AccountName == config.AutoLoginAccount {
+			accountID = *accounts[i].AccountId
+		}
+	}
+
+	roleCredentials, err := ssoClient.GetRoleCredentials(ctx, &sso.GetRoleCredentialsInput{
+		AccessToken: aws.String(credentials.Token),
+		AccountId:   aws.String(accountID),
+		RoleName:    aws.String(config.AutoLoginRole),
+	})
+
+	log.Info("✅ Logged in! Role credentials acquired.", "roleCredentials", roleCredentials.RoleCredentials, "ResultMetadata", roleCredentials.ResultMetadata)
+
+	// Resolve home directory
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		panic(fmt.Errorf("failed to get user home directory: %w", err))
+	}
+
+	// Path to ~/.aws/credentials
+	awsCredentialsPath := filepath.Join(homeDir, ".aws", "credentials")
```

Comment:
You will want to use a go library to read and write AWS config. Package: gopkg.in/ini.v1

We should only manage profiles we have configured. And should leave other configurations untouched.